src/core/fonts.js

```diff
@@ -3351,9 +3352,119 @@ var Font = (function FontClosure() {
       return (charsCache[charsCacheKey] = glyphs);
     },
 
+    getGlyphsAndCharPosition: function Font_getGlyphsAndCharPosition(chars) {
+      const glyphs = [];
+
+      if (this.cMap) {
+        const c = Object.create(null);
+        let i = 0;
+        while (i < chars.length) {
+          this.cMap.readCharCode(chars, i, c);
+          const length = c.length;
+          const isSpace = length === 1 && chars.charCodeAt(i) === 0x20;
+          const glyph = this._charToGlyph(c.charcode, isSpace);
+          glyphs.push({
+            start: i,
+            end: i + length,
+            glyph,
+          });
+          i += length;
+        }
+      } else {
+        for (let i = 0, ii = chars.length; i < ii; ++i) {
+          const charcode = chars.charCodeAt(i);
+          const glyph = this._charToGlyph(charcode, charcode === 0x20);
+          glyphs.push({
+            start: i,
+            end: i + 1,
+            glyph,
+          });
+        }
+      }
+
+      return glyphs;
+    },
+
+    getCharPosition: function Font_getCharPosition(chars) {
+      // Chars can have different sizes
+      // so get the char positions in the string
+      // it's useful to split a string at correct char positions.
```

Comment:
This isn't all that easy to ready, since it feels like you're using line-breaks instead of proper punctuation here.
Also, should this information perhaps be a JSDoc comment instead?

Reply:
Also, you probably want to add a *lazily* initialized cache for the data (something similar to `charsToGlyphs` above), to avoid unnecessary re-parsing here.